components/registry-facade/pkg/blobserve/blobserve.go

```diff
@@ -185,14 +185,18 @@ func (reg *Server) blobFor(ctx context.Context, ref string, readOnly bool) (fs h
 	dgst := reg.refcache[ref]
 	reg.mu.RUnlock()
 
-	if dgst == "" {
+	state := blobUnknown
```

Comment:
The idea of `refcache` was to serve as index for ready blobs.
That is to say, if we hit refcache, the state will be `blobReady` (or it was, at least once).
There is the assumption here that once a blob was downloaded, it's available to use indefinitely (i.e. manually clearing out the blobspace alone does not "remove" the blob, but creates an inconsistent state).
It was not the best way to do that because it assumes the responsibility of the blobspace (it's the blobspace that tells us if the blob is ready or not).

nit: That said, I found the previous version of this function easier to follow.

Reply:
discussed that Chris will look in the proper solution while working on blob download synchronization